dnf/rpm/miscutils.py

```diff
@@ -30,36 +61,41 @@ def checkSig(ts, package):
     return 3 if the key is not trusted
     return 4 if the pkg is not gpg or pgp signed"""
 
-    value = 0
+    value = 4
     currentflags = ts.setVSFlags(0)
     fdno = os.open(package, os.O_RDONLY)
     try:
         hdr = ts.hdrFromFdno(fdno)
     except rpm.error as e:
         if str(e) == "public key not available":
             value = 1
-        if str(e) == "public key not trusted":
+        elif str(e) == "public key not trusted":
             value = 3
-        if str(e) == "error reading package header":
+        elif str(e) == "error reading package header":
             value = 2
+        else:
+            raise ValueError('Unexpected error value %r from ts.hdrFromFdno when checking signature.' % str(e))
     else:
         # checks signature from an hdr
         string = '%|DSAHEADER?{%{DSAHEADER:pgpsig}}:{%|RSAHEADER?{%{RSAHEADER:pgpsig}}:' \
                  '{%|SIGGPG?{%{SIGGPG:pgpsig}}:{%|SIGPGP?{%{SIGPGP:pgpsig}}:{(none)}|}|}|}|'
         try:
             siginfo = hdr.sprintf(string)
             siginfo = ucd(siginfo)
+            rpm_pgpsig_format_regex = re.compile(r'[0-9]{2}:[0-9]{2}:[0-9]{2} [0-9]{4}, Key ID [0-9a-f]{16}\Z')
+
             if siginfo == '(none)':
                 value = 4
+            elif rpm_pgpsig_format_regex.search(siginfo) and _verifyPkgUsingRpmkeys(package, ts.ts.rootDir):
```

Comment:
This is good enough for now, but it is better to do the check in the reverse order: invoke `rpmkeys --checksig` before parsing the package.